server_tgi.py

```diff
@@ -0,0 +1,284 @@
+            # Break from for loop after the first tool_call is streamed if functions is provided
+            if functions and tool_call_count == 2:
+                break
+        yield "data: [DONE]\n\n"
+
+    if request.stream:
+        return StreamingResponse(
+            completion_stream_generator(
+                tool_choice=tool_func_choice,
+                functions=request.functions,
+            ),
+            media_type="text/event-stream",
+        )
+    else:
+        response = client.text_generation(prompt=prompt, details=True, **hyperparams)
+        # Transformers tokenizers is problematic with some special tokens such that they
+        # are not reflected in `response.generated_text`. Use this hack temporarily first.
+        # Issue: https://github.com/huggingface/text-generation-inference/issues/1984
+        response_text = "".join([token.text for token in response.details.tokens])
+        chat_mess = prompt_template.parse_assistant_response(
+            llm_output=response_text, tool_choice=tool_func_choice
+        )
+        # Convert tool_calls to function_call if request.functions is provided
+        if (
+            request.functions
+            and "tool_calls" in chat_mess
+            and chat_mess["tool_calls"] is not None
+            and len(chat_mess["tool_calls"]) > 0
+        ):
+            chat_mess["function_call"] = {
+                "name": chat_mess["tool_calls"][0]["function"]["name"],
+                "arguments": chat_mess["tool_calls"][0]["function"]["arguments"],
+            }
+            chat_mess["tool_calls"] = None
+
+        # Postprocess finish reason
+        finish_reason = "stop"
+        if "function_call" in chat_mess and chat_mess["function_call"]:
+            finish_reason = "function_call"
+        if "tool_calls" in chat_mess and chat_mess["tool_calls"]:
+            finish_reason = "tool_calls"
+
+        choices = [
+            ChatCompletionResponseChoice(
+                index=0,
+                message=ChatMessage(**chat_mess),
+                finish_reason=finish_reason,
+            )
+        ]
+
+        num_prompt_tokens = len(tokenizer.encode(prompt, add_special_tokens=False))
+        num_generated_tokens = response.details.generated_tokens
+        usage = UsageInfo(
+            prompt_tokens=num_prompt_tokens,
+            completion_tokens=num_generated_tokens,
+            total_tokens=num_prompt_tokens + num_generated_tokens,
+        )
+
+        return ChatCompletionResponse(
+            id=request_id,
+            created=created_time,
+            model=request.model,
+            choices=choices,
+            usage=usage,
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Functionary TGI Server")
+    parser.add_argument(
+        "--model", type=str, default="meetkai/functionary-small-v2.5", help="Model name"
+    )
+    parser.add_argument(
+        "--tgi_endpoint",
```

Comment:
we dont need to use "tgi_" prefix for arguments. Since this script always uses tgi